src/sys/socket/addr.rs

```diff
@@ -794,6 +843,15 @@ impl SockAddr {
             SockAddr::Netlink(NetlinkAddr(ref sa)) => (mem::transmute(sa), mem::size_of::<libc::sockaddr_nl>() as libc::socklen_t),
             #[cfg(any(target_os = "ios", target_os = "macos"))]
             SockAddr::SysControl(SysControlAddr(ref sa)) => (mem::transmute(sa), mem::size_of::<sys_control::sockaddr_ctl>() as libc::socklen_t),
+            #[cfg(any(target_os = "dragonfly",
+                      target_os = "freebsd",
+                      target_os = "ios",
+                      target_os = "macos",
+                      target_os = "netbsd",
+                      target_os = "openbsd",
+                      target_os = "linux",
+                      target_os = "android"))]
+            SockAddr::Ether(_) => unimplemented!()
```

Comment:
I'm not positive what `as_ffi_pair()` is used for, but why can't this be implemented for `Ether`? It needs a comment here explaining why it's unimplemented.

Reply:
@Susurrus `libc::sockaddr` and `libc::sockaddr_dl`/`libc::sockaddr_ll` has diffences size (`mem::size_of`) . i think i can't implement.